polar-core/src/vm.rs

```diff
@@ -1096,48 +1070,43 @@ impl PolarVirtualMachine {
         Ok(())
     }
 
-    fn isa_expr(&mut self, operation: &Operation, left: &Term, right: &Term) -> PolarResult<()> {
+    fn isa_expr(&mut self, left: &Term, right: &Term) -> PolarResult<()> {
         match right.value() {
             Value::Pattern(Pattern::Dictionary(fields)) => {
-                let to_unify = |(field, value): (&Symbol, &Term)| -> Operation {
+                // ? is this tested
+
+                // Produce a constraint like left.field = value
+                let to_unify = |(field, value): (&Symbol, &Term)| -> Term {
+                    let value = self.deref(value);
                     let field = right.clone_with_value(value!(field.0.as_ref()));
                     let left = left.clone_with_value(value!(op!(Dot, left.clone(), field)));
-                    let unify = op!(Unify, left.clone(), value.clone());
-                    match value.value() {
-                        Value::Variable(v) | Value::RestVariable(v) => {
-                            match self.variable_state(v) {
-                                VariableState::Cycle(c) => cycle_constraints(c)
-                                    .clone_with_new_constraint(unify.into_term()),
-                                VariableState::Bound(b) => op!(Unify, left, b),
-                                VariableState::Partial(e) => {
-                                    e.clone_with_new_constraint(unify.into_term())
-                                }
-                                VariableState::Unbound => unify,
-                            }
-                        }
-                        _ => unify,
-                    }
+                    let unify = op!(Unify, left, value);
+                    term!(unify)
                 };
 
-                let mut operation = operation.clone();
-
                 // Add all but the last field constraint to the operation.
```

Comment:
Comment is no longer correct.

How come the so-called "bind dance"  isn't required any more?

Reply:
Not sure, I have no idea what that comment was referring to.